internal/gmm/handler.go

```diff
@@ -1540,17 +1541,31 @@ func AuthenticationProcedure(ue *context.AmfUe, accessType models.AccessType) (b
 	if ausfUri == "" {
 		err = fmt.Errorf("AMF can not select an AUSF by NRF")
 		ue.GmmLog.Errorf(err.Error())
+		gmm_message.SendRegistrationReject(ue.RanUe[accessType], nasMessage.Cause5GMMCongestion, "")
 		return false, err
 	}
 	ue.AusfUri = ausfUri
 
 	response, problemDetails, err := consumer.SendUEAuthenticationAuthenticateRequest(ue, nil)
 	if err != nil {
 		ue.GmmLog.Errorf("Nausf_UEAU Authenticate Request Error: %+v", err)
-		return false, errors.New("Authentication procedure failed")
+		gmm_message.SendRegistrationReject(ue.RanUe[accessType], nasMessage.Cause5GMMCongestion, "")
+		err = fmt.Errorf("Authentication procedure failed")
+		ue.GmmLog.Errorf(err.Error())
+		return false, err
 	} else if problemDetails != nil {
-		ue.GmmLog.Errorf("Nausf_UEAU Authenticate Request Failed: %+v", problemDetails)
-		return false, nil
+		ue.GmmLog.Warnf("Nausf_UEAU Authenticate Request Failed: %+v", problemDetails)
+		var cause uint8
+		switch problemDetails.Status {
+		case 403, 404:
```

Comment:
Better to use:
`case http.StatusForbidden, http.StatusNotFound:`

Reply:
I have fixed.